v2/tools/asoctl/cmd/log.go

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package cmd
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zerologr"
+	"github.com/rs/zerolog"
+	"github.com/vbauerster/mpb/v8"
+)
+
+var (
+	// Global logger and progress bar
+	// While these are global here, they can't be referenced from internal packages and should be passed through
+	// as parameters where needed.
+	log      logr.Logger
+	progress *mpb.Progress
+
+	verbose bool
+)
+
+func init() {
+	// Create Progressbar for console output
+	progress = mpb.New()
```

Comment:
Is there a compelling reason to do this in an `init` function vs some sort of `makeLogger() (logr.Logger, *mpb.Progress)` called in `main`? You're already passing them around everywhere pretty much rather than using them as globals so it seems like it'd be pretty easy to drop the `init` + globals entirely?

Is the issue the `root.go` usage?

Reply:
Want to guarantee they're initialized before use. 

Have deliberately put them into `cmd` so they can't be referenced from `internal` without creating a circular reference, forcing them to be passed in, but this makes it possible for the variables to be directly referenced, bypassing any lazy-initialization.

Can't easily pass them through `cobra` from `root.go`, so this seemed a good compromise.